### ISSUE TITLE
Explore SDL as a GUI option for Graphics Team

### ISSUE THREAD

**Issue by xmvalencia** [uchicago-cs/chiventure#584]
_Sprint Summary:_
In this sprint I'll be looking at SDL as a GUI option instead of raylib. We have run into issues with raylib as it seems to be exclusive to running on Mac (and Windows with excessive tinkering). Right now it's difficult to install and run on Windows and won't be able to run on the CSIL Linux machines. The goal is to research functionality of SDL and see determine it's capability and its compatibility with the linux machines. This is to determine whether we continue development with raylib or switch to SDL.

_Products of the Task:_
Sample programs using SDL that would be useful in chiventure such as tracking mouse movement, manipulating the window, creating shapes for the map and loading images uploaded by the author of the game.

**Comment by xmvalencia** [comment]
_Sprint Update:_
I've been continuing my research into SDL, this means learning how to download, install and implement some functions in SDL. My machine is a Windows machine and from what I've been able to gather, using SDL requires an excess of downloading programs such as Visual Studio (2015) and requires development through that application. I'll be updating this issue with example programs I've written and my analysis on whether to continue development with SDL or Raylib

**Comment by xmvalencia** [comment]
_Sprint Update: Tracking mouse movement, creating shapes and window control_

**Regarding mouse movement:** 
Mouse movement and other mouse driven trigger events such as clicking are recorded by SDL and can be manipulated to perform different tasks. For the following of mouse movement no extra functions must be written but for event triggers minor additions to the code must be added

**Regarding Creating Shapes:** 
As previously discussed against raylib, SDL is pretty low-level when it comes to creating shapes other than rectangles. To create other basic shapes entire modules must be created (for example circles or items). There are resources online that help with building modules for basic shapes, but that means that creating things like our map would require meticulous and often excessive code for what is essentially a lot of rectangles and circles. 

**Regarding Window Control:** 
Window control is loosely tied in with mouse tracking and event triggers. Closing the window, moving the window and resizing the window all seem to be complex functions with SDL. When creating the window we can set the width and length but further manipulation to the code is necessary to resize. When compared to raylib's management of the window, SDL seems to be unnecessarily complicated.


**Image Example:** 
![image](https://user-images.githubusercontent.com/52059114/82841967-be16d980-9e8c-11ea-9471-2be1d3230a5f.png)

**Comment by xmvalencia** [comment]
_Sprint Update: Uploading Images_

**Regarding Uploading Images**
SDL does a mediocre job with managing multiple images at the same time, let alone an extensive array of images that would be necessary for displaying things such as inventory and battles. From what I understand of raylib, it handles multiple image files and functions easier than SDL. I have uploaded some examples of uploading images in the window but they aren't extensive examples as the more you work with SDL the more obvious it becomes that it doesn't handle it as well as raylib. The examples showcase things such as inventory items and loading screens or set backgrounds uploaded by the author

**Image Examples:**
![image](https://user-images.githubusercontent.com/52059114/82845279-bfe69a00-9e98-11ea-9e08-830d1fad59c1.png)
![image](https://user-images.githubusercontent.com/52059114/82845483-a85be100-9e99-11ea-8df8-c675c2633cb0.png)

**Comment by xmvalencia** [comment]
_Sprint Purpose and Summary:_
During this sprint I looked at SDL as a GUI option instead of raylib. We have run into issues with raylib as it seems to be exclusive to running on Mac (and Windows with excessive tinkering). We had done initial research and decided that raylib was the more extensive library that might be a better fit for the project we were working on, but with the amount of problems we ran while developing with raylib we decided to take a step back and look at possibly switching to SDL. The goal was to research functionality of SDL and see determine if we should switch to libSDL.

_Products of the Task:_
I have produced sample programs using SDL that would be useful in chiventure such as tracking mouse movement, manipulating the window, creating shapes for the map and loading images uploaded by the author of the game. This was to give us an idea of how we could use SDL, if we deemed that it was better than continuing our development with raylib.

**Final Conclusions:**
This issue was run in conjuction with the continuous development with raylib by my teammates. Hearing and their progress with making raylib work on linux machines and understanding the abilities of SDL to build programs that would help the graphics team it's clear that we should continue development with raylib and NOT switch to SDL as a graphics library
